docs/source/webhooks.rst

```diff
@@ -0,0 +1,87 @@
+Webhooks
+========
+
+Webhooks allow you to integrate external systems with |st2| using HTTP
+webhooks. Unlike sensors which use "pull" approach, webhooks use "push"
+approach. This means they work by you pushing triggers directly to the |st2|
+API using HTTP POST request.
+
+What is a difference between sensors and webhooks?
+--------------------------------------------------
+
+Sensors integrate with external systems and services using poll approach
+(sensors periodically reach out to an external system to retrieve data you are
+interested in) and webhooks use push approach (your systems push data to the
+|st2| API when an event you are interested in occurs).
+
+Sensors are a preferred way of integration since they offer a more granular and
+tighter integration.
+
+On the other hand, webhooks come handy when you have an existing script or
+software which you can easily modify to send a webhook to the |st2| API when an
+event you are interested in occurs.
+
+Another example where webhooks come handy is when you want to consume events
+from a 3rd party services which already offer webhooks integration - an example
+of such service includes Github.
+
+Authentication
+--------------
+
+All the requests to the /webhooks endpoints needs to be authenticated in the
+same way as other API requests. This means the requests need to contain
+``X-Auth-Token`` header with a valid authentication token.
```

Comment:
I'm actually talking on our end? where is `matoken` set in Stackstorm? In the rule?